src/test/java/org/onebusaway/alexa/AuthedSpeechletTest.java

```diff
@@ -290,28 +299,7 @@ public void repeat() throws SpeechletException, URISyntaxException, IOException
         // Try the repeat intent - this simulates after initial setup, when we don't yet have anything to repeat
         SpeechletResponse sr = authedSpeechlet.onIntent(repeatIntent, session);
         String spoken = ((PlainTextOutputSpeech)sr.getOutputSpeech()).getText();
-        assertThat(spoken, containsString("I'm sorry, I don't have anything to repeat.  You can ask me for arrival times for your stop."));
-
-        ObaArrivalInfo[] obaArrivalInfoArray = new ObaArrivalInfo[1];
-        obaArrivalInfoArray[0] = obaArrivalInfo;
-        new Expectations() {{
-            obaArrivalInfo.getShortName(); result = "8";
-            obaArrivalInfo.getHeadsign(); result = "Mlk Way Jr";
-            obaArrivalInfoResponse.getArrivalInfo(); result = obaArrivalInfoArray;
-            obaUserClient.getArrivalsAndDeparturesForStop(anyString, anyInt); result = obaArrivalInfoResponse;
-        }};
-        String response = "Route 8 Mlk Way Jr is now arriving based on the schedule -- ";
-
-        // Test initial request/response
-        sr = authedSpeechlet.onLaunch(launchRequest, session);
-        spoken = ((PlainTextOutputSpeech)sr.getOutputSpeech()).getText();
-        assertThat(spoken, equalTo(response));
-
-        // Now try repeat intent again - we should get the last response again
-        session.setAttribute(PREVIOUS_RESPONSE, response);  // FIXME - this should be set via the above call, but isn't?
-        sr = authedSpeechlet.onIntent(repeatIntent, session);
-        spoken = ((PlainTextOutputSpeech)sr.getOutputSpeech()).getText();
-        assertThat(spoken, containsString(response));
+        assertThat(spoken, equalTo(prevResponse));
```

Comment:
I don't think that this is effectively any different than the last assertion in my initial repeat test.

I was trying to assert 3 items:

1. Upon initial setup, before any getCity, getStop, or getArrivals intents are called, we should return the "sorry, nothing to repeat" message.
2. Simulate the user triggering the getArrivals intent - assert that we get the expected arrival response back
3. Trigger the repeat intent - the repeated response should match the number 2 response above.

The issue in my code is that I shouldn't need to call:
`session.setAttribute(PREVIOUS_RESPONSE, response);`

The session attribute should be set via:
`sr = authedSpeechlet.onLaunch(launchRequest, session);`

...but for some reason it's not.

In your commit, you're also directly setting `session.setAttribute(PREVIOUS_RESPONSE, response);` - I was trying to avoid this to fully simulate setting the previously response from actual code (instead of in test).

Thoughts on how to change this so the response text is actually generated from within the AuthedSpeechlet?